runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -10363,82 +10340,66 @@ genHeapAlloc(TR::Node * node, TR::Instruction *& iCursor, bool isVariableLen, TR
-                  iCursor = generateRSInstruction(cg, TR::InstOpCode::SRA, node, shiftReg, 8, iCursor);
+                  iCursor = generateRRInstruction(cg, TR::InstOpCode::LR, node, loopCountRegister, sizePreserveRegister, iCursor);
+                  iCursor = generateRSInstruction(cg, TR::InstOpCode::SRA, node, loopCountRegister, 8, iCursor);
                   }
                }
 
+            TR::Register * addressReg = srm->findOrCreateScratchRegister();
+            iCursor = generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, addressReg, resReg, iCursor);
+
+            TR::LabelSymbol * ResidueBytesZeroingLabel = generateLabelSymbol(cg);
+            iCursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BZ, node, ResidueBytesZeroingLabel, iCursor);
             // Zero blocks of 256 bytes if size > 256.
-            iCursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BZ, node, fillerRemLabel, iCursor);
-            iCursor = generateS390LabelInstruction(cg, TR::InstOpCode::label, node, fillerLoopLabel);
-            iCursor = generateSS1Instruction(cg, TR::InstOpCode::XC, node, 255, generateS390MemoryReference(addressReg, 0, cg), generateS390MemoryReference(addressReg, 0, cg), iCursor);
-            iCursor = generateRXInstruction(cg, TR::InstOpCode::LA, node, addressReg, generateS390MemoryReference(addressReg, 256, cg), iCursor);
-            iCursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRCT, node, shiftReg, fillerLoopLabel);
-            iCursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, fillerRemLabel);
-
-            // Zero the residue bytes.
+            genMemoryZeroingLoop(cg, node, iCursor, addressReg, loopCountRegister);
+
+            // Zero residue bytes.
             TR::LabelSymbol *exrlTargetLabel = generateLabelSymbol(cg);
+            iCursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, ResidueBytesZeroingLabel);
             iCursor = generateS390LabelInstruction(cg, TR::InstOpCode::label, node, exrlTargetLabel);
             iCursor = generateSS1Instruction(cg, TR::InstOpCode::XC, node, 0, generateS390MemoryReference(addressReg, 0, cg), generateS390MemoryReference(addressReg, 0, cg), iCursor);
-            iCursor = generateS390LabelInstruction(cg, TR::InstOpCode::label, node, fillerRemLabel);
-            iCursor = generateRILInstruction(cg, TR::InstOpCode::EXRL, node, lengthReg, exrlTargetLabel);
+            iCursor = generateS390LabelInstruction(cg, TR::InstOpCode::label, node, ResidueBytesZeroingLabel);
+            iCursor = generateRILInstruction(cg, TR::InstOpCode::EXRL, node, sizePreserveRegister, exrlTargetLabel);
+
+            srm->reclaimScratchRegister(addressReg);
+            srm->reclaimScratchRegister(sizePreserveRegister);
             }
          else
             {
+            int32_t loops = allocSize / 256;
```

Comment:
May be this may have escaped previous PR, but loops sounds like number of loop you will be generating. May be `loopIterCount` or something synonymous can be used here?